File: Project/backend/app/models/utente.py
import enum
from typing import Optional
from sqlmodel import Field, SQLModel

class MezzoSpostamento(str, enum.Enum):
    A_PIEDI = "A PIEDI"
    BICI_A_NOLEGGIO = "BICI A NOLEGGIO"
    AUTOBUS = "AUTOBUS"
    TRENO = "TRENO"
    MOTO = "MOTO"
    AUTO = "AUTO"
    ALTRO = "ALTRO"

class UtenteBase(SQLModel):
    nome: str = Field(max_length=100, index=True)
    cognome: str = Field(max_length=100, index=True)
    campus: str | None = Field(default=None, max_length=100)
    mezzo_di_spostamento: MezzoSpostamento | None = Field(default=MezzoSpostamento.A_PIEDI)

class Utente(UtenteBase, table=True):
    __tablename__ = "utente"
    id: int | None = Field(default=None, primary_key=True)
    mezzo_di_spostamento: str | None = Field(default=MezzoSpostamento.A_PIEDI.value, max_length=50)

class UtenteCreate(UtenteBase):
    pass

class UtentePublic(UtenteBase):
    id: int

class UtenteUpdate(UtenteBase):
    nome: str | None = Field(default=None, max_length=100)
    cognome: str | None = Field(default=None, max_length=100)
    campus: str | None = Field(default=None, max_length=100)
    mezzo_di_spostamento: MezzoSpostamento | None = Field(default=None)
    

File: Project/ContextAware_app/app/src/main/java/com/example/myapplication/data/model/utente.kt
package com.example.myapplication.data.model
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class UtentePublic(
    val id: Int,
    val nome: String,
    val cognome: String,
    val campus: String? = null,
    @SerialName("mezzo_di_spostamento")
    val mezzoDiSpostamento: MezzoSpostamento? = MezzoSpostamento.A_PIEDI
)

@Serializable
data class UtenteCreate(
    val nome: String,
    val cognome: String,
    val campus: String? = null,
    @SerialName("mezzo_di_spostamento")
    val mezzoDiSpostamento: MezzoSpostamento? = MezzoSpostamento.A_PIEDI,
    val password: String
)


@Serializable
data class UtenteUpdate(
    val nome: String? = null,
    val cognome: String? = null,
    val campus: String? = null,
    @SerialName("mezzo_di_spostamento")
    val mezzoDiSpostamento: MezzoSpostamento? = null
)
